airflow/www_rbac/views.py

```diff
@@ -278,27 +290,33 @@ def dag_stats(self, session=None):
             session.query(ds.dag_id, ds.state, ds.count)
         )
 
-        data = {}
-        for dag_id, state, count in qry:
-            if dag_id not in data:
-                data[dag_id] = {}
-            data[dag_id][state] = count
+        filter_dag_ids = appbuilder.sm.get_accessible_dag_ids()
 
         payload = {}
-        for dag in dagbag.dags.values():
-            payload[dag.safe_dag_id] = []
-            for state in State.dag_states:
-                try:
-                    count = data[dag.dag_id][state]
-                except Exception:
-                    count = 0
-                d = {
-                    'state': state,
-                    'count': count,
-                    'dag_id': dag.dag_id,
-                    'color': State.color(state)
-                }
-                payload[dag.safe_dag_id].append(d)
+        if filter_dag_ids:
+            if filter_dag_ids != {'all_dags'}:
```

Comment:
Again `'all_dags' in filter_dag_ids` is safer, if a user ends up with both `all_dags` and `some_dag_id` they'll only have access to `filter_dag_ids`.

Reply:
updated. thanks.